### PR TITLE
Update <a> ping attribute

### LINKED ISSUE #9219: html.elements.a - Ping attribute supported by Safari
#### What information was incorrect, unhelpful, or incomplete?

The compatibility data states that Safari does not support the `ping` attribute on anchor tags. The `ping` attribute seems to be supported by Safari according to https://caniuse.com/ping and when I created a small test locally.

#### What did you expect to see?

Safari to be supported instead it says No.

#### Did you test this? If so, how?

1. Added an anchor tag with `ping` attribute
2. Clicked on it in Safari (v14.0.3)
3. Checked network tab in Web Inspector, shows network request to ping url.

<details>
<summary>MDN page report details</summary>

* Query: `html.elements.a`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a
* Report started: 2021-02-22T09:27:05.147Z

</details>

### PR DESCRIPTION
**Pull request by ddbeck**
Tested by clicking a link with `ping` set to a server running locally and inspecting the requests. First works in Safari 6 (both platforms).

Fixes #9219.

### CODE REVIEW

**Comment by foolip** [comment]
https://developer.mozilla.org/en-US/docs/Web/API/HTMLAnchorElement#browser_compatibility says 5.1 / 5, can you make them match?

It's possible the content attribute and IDL attribute didn't ship at the same time, but not very common for the IDL attribute to ship first.

**Comment by ddbeck** [comment]
@foolip 

> It's possible the content attribute and IDL attribute didn't ship at the same time, but not very common for the IDL attribute to ship first.

Unfortunately, I think this might be one of those cases. The collector test shows `true` on Safari 5.1 but not before.

Searching the source for HTMLAnchorElement ping feature made me notice the equivalent `<area>` feature, which sent me down a rabbit hole. Some things I learned/did:

- The anchor `ping` feature was wrongly set as experimental. Fixed.
- `<area>` was missing a `ping` feature. I added it (to correspond to `HTMLAreaElement`) and checked that the data was reasonable.
- I confirmed the truthiness of the feature for Chrome and Firefox (and the pref, for Firefox), but I wasn't able to determine the first version for either. They've both got it as far back as SauceLabs and BrowserStack will take me.

**Comment by ddbeck** [comment]
Hey @foolip, did you have any more thoughts on the unusual IDL situation here?

**Comment by foolip** [comment]
I went ahead and found the two WebKit commits that introduced the content attribute and the IDL attribute.

`<a ping>` was added but disabled by default at WebKit trunk version 534.9.0:
https://trac.webkit.org/changeset/68166/webkit
https://trac.webkit.org/browser/webkit/trunk/WebCore/Configurations/Version.xcconfig?rev=68166

The IDL attribute was added in WebKit trunk version 534.27.0:
https://trac.webkit.org/changeset/82499/webkit
https://trac.webkit.org/browser/webkit/trunk/Source/WebCore/Configurations/Version.xcconfig?rev=82499

But because the feature was disabled by default, testing is necessary. I also wrote a test to confirm Safari 6:
```html
<a href="/bla.html" ping="https://foolip.org/pinganchor">leave me</a>
<map name="map">
  <area shape="rect" coords="0,0,100,100"
        href="/bla.html"
        ping="https://foolip.org/pingarea">
</map>
<img usemap="#map" src="bla.png">
```

Other things I noticed when looking at all "ping" features:
- Edge data is implausible, with SVG `ping` supposedly added before HTML
- Firefox support inconsistently behind a flag or not
- Missing Opera for Android data

But I'll focus only on what's outright wrong in this PR.

File: html/elements/area.json [hunk 1]
```diff
@@ -520,6 +520,67 @@
...
+          "__compat": {
+            "support": {
+              "chrome": {
+                "version_added": true
+              },
+              "chrome_android": {
+                "version_added": true
+              },
+              "edge": {
+                "version_added": "79"
```
Comment (foolip) [inline html/elements/area.json:533]:
```suggestion
                "version_added": "17"
```

api.HTMLAreaElement.ping says "17" and I can confirm it using http://mdn-bcd-collector.appspot.com/tests/api/HTMLAreaElement/ping. With my test in https://github.com/mdn/browser-compat-data/pull/9470#issuecomment-817091562 I can also confirm the behavior for both `<a ping` and `<area ping>` was introduced in Edge 17.

**Review by foolip — CHANGES_REQUESTED** [review]

File: html/elements/area.json (hunk 1 above)
Comment (ddbeck) [reply html/elements/area.json]: Fixed with 1878c56dd.

File: html/elements/area.json [hunk 2]
```diff
@@ -520,6 +520,67 @@
...
+                  {
+                    "type": "preference",
+                    "name": "browser.send_pings",
+                    "value_to_set": "true"
+                  }
+                ]
+              },
+              "firefox_android": {
+                "version_added": true,
+                "flags": [
```
[the remark this answers is not in the document]
Comment (ddbeck) [reply html/elements/area.json:548]: Fixed with a71870d3c (sort of, more to the Firefox story coming shortly).

File: html/elements/area.json [hunk 3]
```diff
@@ -520,6 +520,67 @@
...
+                ]
+              },
+              "ie": {
+                "version_added": false
+              },
+              "opera": {
+                "version_added": true
+              },
+              "opera_android": {
+                "version_added": null
```
[the remark this answers is not in the document]
Comment (ddbeck) [reply html/elements/area.json]: Fixed with 9c9baf959.

**Comment by ddbeck** [comment]
> Edge data is implausible, with SVG ping supposedly added before HTML

I didn't even think of SVG at all. 🤦  I guess I need to work through that completely.

> Firefox support inconsistently behind a flag or not

OK, quite a bit of time elapsed from me researching this to now, so it took me a bit time to figure out what I was thinking. The story is this: Firefox has the attribute as part of the API without a flag, but never actually sends the ping without the flag. I guess I had in my head that, since the ping happens on click, that's a behavior of the content attribute? I'm not sure that's a great way to put it. Perhaps it'd be better to do something like two support statements (for both element and API), one with `partial_implementation` and a note saying the ping is never sent, and second with the flag for full support?

> Missing Opera for Android data

I think this is fixed? Or at least it's better than it was.

**Comment by ddbeck** [comment]
OK, I've corrected the SVG `<a>` element `ping` attribute data:

- Firefox's support is behind a flag. For consistency, I've shown this the way the other Firefox `ping` data is (with no flag for the API). I can change this all in one go, if we can decide on a way to talk about this.
- Edge never supported the attribute in SVG.

**Comment by foolip** [comment]
> I guess I had in my head that, since the ping happens on click, that's a behavior of the content attribute? I'm not sure that's a great way to put it.

As implemented, it is a behavior of the content attribute, the IDL attribute only gets and sets the content attribute. However, when the two kinds of attributes come apart it's almost always a bug and potentially confusing, and if the schema somehow allowed I think this should be considered one feature, with `partial_implementation` for when only one half worked.

> Perhaps it'd be better to do something like two support statements (for both element and API), one with `partial_implementation` and a note saying the ping is never sent, and second with the flag for full support?

When support for the IDL attribute comes before the content attributes, indeed `partial_implementation` for the initial support of the IDL attributes seems good. The other way around also happened here, but it doesn't seem so important to point out that the content attribute worked even earlier, since it's so distant now.

**Review by foolip — APPROVED** [review]

**Comment by ddbeck** [comment]
> since it's so distant now

Yes, agreed. I'll merge this now. Thank you for making this one a lot better than it was at the start. 👍

**Pull request merged**